dev/com.ibm.ws.jpa.tests.jpa_32_fat/test-applications/jakartadata/src/io/openliberty/jpa/data/tests/web/JakartaDataRecreateServlet.java

```diff
@@ -168,6 +168,52 @@ public void testOLGH28913() throws Exception {
 
         assertEquals(character.getHexadecimal(), result);
     }
+    
+    @Test
+    @Ignore("Reference issue:https://github.com/OpenLiberty/open-liberty/issues/29459")
+    public void testOLGH29459() throws Exception {
+        // Sample segment ID and points
+        int x1 = 0, y1 = 0, x2 = 120, y2 = 209;
+        
+        // Create a new segment using the 'of' factory method
+        Segment segment = Segment.of(x1, y1, x2, y2);
+        
+        // Begin the transaction
+        tx.begin();
+
+        try {
+            // Persist the segment using EntityManager
+            em.persist(segment);
+
+            // Insert into the database using a native SQL query
+            em.createNativeQuery("INSERT INTO Segment (id, pointA_x, pointA_y, pointB_x, pointB_y) VALUES (?, ?, ?, ?, ?)")
+                .setParameter(1, segment.id)      // Segment ID
+                .setParameter(2, segment.pointA.x())  // Point A X
+                .setParameter(3, segment.pointA.y())  // Point A Y
+                .setParameter(4, segment.pointB.x())  // Point B X
+                .setParameter(5, segment.pointB.y())  // Point B Y
+                .executeUpdate();
+
+            // Commit the transaction
+            tx.commit();
+        } catch (Exception e) {
+            // If there's any exception, rollback the transaction
+            tx.rollback();
+            throw e;
+        }
```

Comment:
```suggestion
        try {
            tx.being();

            // Insert into the database using a native SQL query
            em.createNativeQuery("INSERT INTO Segment (id, pointA_x, pointA_y, pointB_x, pointB_y) VALUES (?, ?, ?, ?, ?)")
                .setParameter(1, segment.id + 1)      // Segment ID
                .setParameter(2, segment.pointA.x())  // Point A X
                .setParameter(3, segment.pointA.y())  // Point A Y
                .setParameter(4, segment.pointB.x())  // Point B X
                .setParameter(5, segment.pointB.y())  // Point B Y
                .executeUpdate();

            // Commit the transaction
            tx.commit();
        } catch (Exception e) {
            // If there's any exception, rollback the transaction
            tx.rollback();
            // Expect an exception here if EclipseLink did not create the column names using the pattern
            //     [embedded_entity_field_name]_[record_field_name]
            throw e;
        }
```

Reply:
This query will fail as is because you attempting to insert an entity with the same ID as the entity you persisted with `em.persist()`.  Instead, we can increment the Id by 1, and add in a comment about what we are testing (that the column names are generated in a known pattern).